### PR TITLE
chore(sdk): Add adapter type for `NodePrimitives::Receipt`

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/paradigmxyz/reth/issues/12454

### CODE REVIEW

File: crates/primitives-traits/src/node.rs [hunk 1]
```diff
@@ -73,3 +73,6 @@ where
     type TxType = T::TxType;
     type Receipt = T::Receipt;
 }
+
+/// Helper adapter type for accessing [`NodePrimitives`] receipt type.
+pub type Rcpt<N> = <N as NodePrimitives>::Receipt;
```
Comment (mattsse) [inline crates/primitives-traits/src/node.rs:78]: not a fan of abbreviations for aliases.

we will likely need a few more, so I think we should come up with a common prefix for these, perhaps just `NodeReceipt`?

Comment (emhane) [reply crates/primitives-traits/src/node.rs:78]: I thought we rename it when we don't use `reth_primitives::Receipt` in the code base anymore?

Comment (emhane) [reply crates/primitives-traits/src/node.rs:78]: ah no the trait is the problem nvm...wanted smthg short...

Comment (emhane) [reply crates/primitives-traits/src/node.rs:78]: `ReceiptTy`? common suffix?

File: crates/primitives-traits/src/node.rs [hunk 2]
```diff
@@ -73,3 +73,6 @@ where
     type TxType = T::TxType;
     type Receipt = T::Receipt;
 }
+
+/// Helper adapter type for accessing [`NodePrimitives`] receipt type.
+pub type ReceiptTy<N> = <N as NodePrimitives>::Receipt;
```
Comment (mattsse) [inline crates/primitives-traits/src/node.rs:78]: I can live with a common suffix

**Review by mattsse — APPROVED** [review]

**Pull request merged**